project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module

```diff
@@ -161,10 +161,214 @@ function c4m_og_menu() {
+ * with action = 'reject'.
+ *
+ * @param object $form
+ *   Form array to populate.
+ * @param object $form_state
+ *   Form state array.
+ */
+function c4m_og_reject_membership_request_submit($form, &$form_state) {
+  $membership = $form_state['membership'];
+  $account = $form_state['account'];
+  $gid = $form_state['values']['gid'];
+  $timestamp = $form_state['values']['timestamp'];
+  $hashed_pass = $form_state['values']['hashed_pass'];
+
+  c4m_og_control_membership('reject', $gid, $account, $membership, $timestamp, $hashed_pass);
+}
+
+/**
+ * Function builds and returns a link for membership request control.
+ *
+ * Includes a token for auto login.
+ *
+ * @param object $account
+ *   Account which performs the action.
+ * @param object $membership
+ *   Membership for which action is performed.
+ * @param string $action
+ *   Approve / Reject.
+ *
+ * @return string
+ *   A link for membership request control.
+ */
+function c4m_og_autologin_control_membership_link($account, $membership, $action) {
+  $timestamp = REQUEST_TIME;
+  $options = array(
+    'absolute' => TRUE,
+    'purl' => array('disabled' => TRUE),
+  );
+
+  return url('group/node/' . $membership->gid . '/admin/membership-' . $action . '/' . $account->uid . '/'
+    . $membership->id . '/' . $timestamp . '/' . user_pass_rehash($account->pass, $timestamp, $account->login, $account->uid), $options);
+}
+
+/**
+ * Retrieves group ID from form and forwards to it's People page.
+ *
+ * @param object $form
+ *   Form array to populate.
+ * @param object $form_state
+ *   Form state array.
+ */
+function c4m_og_route_to_group_people($form, &$form_state) {
+  $gid = $form_state['values']['gid'];
+
+  drupal_goto('group/node/' . $gid . '/admin/people');
+}
+
+/**
+ * Function approves / rejects membership request, submitted by user.
+ *
+ * Verifies auto login token, and if valid, logs in the administrator and
+ * performs required state change to membership request. If not valid, forwards
+ * to login page.
+ *
+ * @param string $action
+ *   Approve / Reject.
+ * @param int $gid
+ *   Group ID.
+ * @param object $account
+ *   Administrator user that performs the action.
+ * @param object $membership
+ *   OG membership on which action is performed.
+ * @param int $timestamp
+ *   Timestamp of the moment action was requested.
+ * @param string $hashed_pass
+ *   Validation for executing user.
+ */
+function c4m_og_control_membership($action, $gid, $account, $membership, $timestamp = 0, $hashed_pass = '') {
+  global $user;
+  // Time out in seconds, until auto login token expires.
+  // 24 hours = 86400 seconds.
+  $timeout = 86400;
+  $current = REQUEST_TIME;
+
+  // If not logged in, or logged in as different user.
+  if (user_is_anonymous() || $account->uid != $user->uid) {
+    // Validate expiration and hashed password/login.
+    if ($timestamp <= $current && $current - $timestamp < $timeout && $account->uid && $timestamp >= $account->login &&
```

Comment:
@anvmn 
please extract each condition to a variable so it will be easier to read it, thanks.